geode-core/src/main/java/org/apache/geode/internal/cache/TXCommitMessage.java

```diff
@@ -1384,153 +1323,145 @@ protected void txApplyEntryOpAdjunctOnly(FarSideEntryOp entryOp) {
-        DataSerializer.writeString(this.internalRegion.getFullPath(), out);
-        if (this.internalRegion instanceof BucketRegion) {
+      if (internalRegion != null) {
+        DataSerializer.writeString(internalRegion.getFullPath(), out);
+        if (internalRegion instanceof BucketRegion) {
           DataSerializer.writeString(
-              ((Bucket) this.internalRegion).getPartitionedRegion().getFullPath(), out);
+              ((Bucket) internalRegion).getPartitionedRegion().getFullPath(), out);
         } else {
           DataSerializer.writeString(null, out);
         }
       } else {
-        DataSerializer.writeString(this.regionPath, out);
-        DataSerializer.writeString(this.parentRegionPath, out);
+        DataSerializer.writeString(regionPath, out);
+        DataSerializer.writeString(parentRegionPath, out);
       }
 
-      if (isEmpty() || this.opKeys.size() == 0) {
+      if (isEmpty() || opKeys.size() == 0) {
         out.writeInt(0);
       } else {
-        int size = this.opKeys.size();
+        int size = opKeys.size();
         out.writeInt(size);
 
         final boolean largeModCount;
-        if (this.msg.txState != null) {
-          largeModCount = this.msg.txState.needsLargeModCount();
+        if (msg.txState != null) {
+          largeModCount = msg.txState.needsLargeModCount();
         } else {
-          largeModCount = this.msg.needsLargeModCount;
+          largeModCount = msg.needsLargeModCount;
         }
         out.writeBoolean(largeModCount);
 
-        final boolean sendVersionTags =
-            this.msg.clientVersion == null
-                || KnownVersion.GFE_70.compareTo(this.msg.clientVersion) <= 0;
-        if (sendVersionTags) {
-          VersionSource member = this.memberId;
-          if (member == null) {
-            if (this.internalRegion == null) {
-              Assert.assertTrue(this.msg.txState == null);
-            } else {
-              member = this.internalRegion.getVersionMember();
-            }
+        VersionSource member = memberId;
+        if (member == null) {
+          if (internalRegion == null) {
+            Assert.assertTrue(msg.txState == null);
+          } else {
+            member = internalRegion.getVersionMember();
           }
-          DataSerializer.writeObject(member, out);
         }
+        DataSerializer.writeObject(member, out);
         for (int i = 0; i < size; i++) {
-          DataSerializer.writeObject(this.opKeys.get(i), out);
-          if (this.msg.txState != null) {
+          DataSerializer.writeObject(opKeys.get(i), out);
+          if (msg.txState != null) {
             /* we are still on tx node and have the entry state */
-            ((TXEntryState) this.opEntries.get(i)).toFarSideData(out, context, largeModCount,
-                sendVersionTags, useShadowKey);
+            ((TXEntryState) opEntries.get(i)).toFarSideData(out, context, largeModCount,
+                true, useShadowKey);
```

Comment:
Now sendVersionTags is always set to to true irrespective of if msg.clientVersion is null or not.

Reply:
For the same reason as above.